### PR TITLE
additional CSS selector formatting

### PR DESCRIPTION
**Pull request by flavorjones**
Implement CSS selector formatting for IdSelector, HashToken, PseudoClassSelector, and PseudoElementSelector, and add test coverage.

Also backfill coverage of the existing PseudoElementSelector AST.

### CODE REVIEW

File: lib/syntax_tree/css/selectors.rb [hunk 1]
```diff
@@ -125,7 +125,7 @@ def accept(visitor)
         end
 
         def child_nodes
-          [type, subclasses, pseudo_elements].flatten
+          [type, subclasses, pseudo_elements].compact.flatten
```
Comment (kddnewton) [inline lib/syntax_tree/css/selectors.rb:128]: Up to you, but you could make this:

```suggestion
          [type, *subclasses, *pseudo_elements]
```

**Review by kddnewton — APPROVED** [review]

File: lib/syntax_tree/css/selectors.rb (hunk 1 above)
Comment (flavorjones) [reply lib/syntax_tree/css/selectors.rb:128]: Unfortunately that change doesn't account for a) `type` being nil and needing to be removed, and b) `pseudo_elements` contains nested arrays.

Comment (flavorjones) [reply lib/syntax_tree/css/selectors.rb:128]: While we're here, I don't really like the fact that pseudo_elements contains nested arrays (see the test "parses a compound selector with pseudo-elements and pseudo-classes" to see what it looks like), but regardless I think `child_nodes` should flatten it all.

**Comment by flavorjones** [comment]
@kddnewton If this is OK with you, would you mind merging? I've got more work that's based on this branch. :heart:

**Pull request merged**